docs/reference/commandline/run.md

```diff
@@ -186,6 +186,64 @@ flag makes Docker attempt to create a new file and write the container ID to it.
 If the file exists already, Docker returns an error. Docker closes this
 file when `docker run` exits.
 
+### <a name="pid"></a> PID settings (--pid)
+
+```text
+--pid=""  : Set the PID (Process) Namespace mode for the container,
+             'container:<name|id>': joins another container's PID namespace
+             'host': use the host's PID namespace inside the container
+```
+
+By default, all containers have the PID namespace enabled.
+
+PID namespace provides separation of processes. The PID Namespace removes the
+view of the system processes, and allows process ids to be reused including
+PID 1.
+
+In certain cases you want your container to share the host's process namespace,
+basically allowing processes within the container to see all of the processes
+on the system. For example, you could build a container with debugging tools
+like `strace` or `gdb`, but want to use these tools when debugging processes
+within the container.
+
+#### Example: run htop inside a container
+
+Create this Dockerfile:
+
+```dockerfile
+FROM alpine:latest
+RUN apk add --update htop && rm -rf /var/cache/apk/*
+CMD ["htop"]
+```
+
+Build the Dockerfile and tag the image as `myhtop`:
+
+```console
+$ docker build -t myhtop .
+```
+
+Use the following command to run `htop` inside a container:
+
+```console
+$ docker run -it --rm --pid=host myhtop
+```
+
+#### Example, join another container's PID namespace
+
+Joining another container's PID namespace can be useful for debugging that
+container. Start a container running a Redis server:
+
+```console
+$ docker run --name my-redis -d redis
```

Comment:
"note to self"; see how many `redis` examples we have, and whether we should consider `nginx` / `nginx:alpine` for these if we don't really need `redis`.

Reply:
Agree, we need to look into this at a broader level, but I'm changing this one to nginx for now